From 17ed363319d47aa2b894f40c4c2c3eb0c1ec8f84 Mon Sep 17 00:00:00 2001
From: Charitardha <satya.charitardha.jayanti@intel.com>
Date: Mon, 6 Aug 2012 17:42:06 +0530
Subject: Audio loss when playing .mp4 clip with mp3 audio

BZ: 50758

Setting correct mime type according to the object type
indication in MPEG4Extractor will fix the issue.

Category: feature diffrentiation
Domain: AUDIO_MEDIA-Audio Codecs
Origin: Internal
Upstream-Candidate: no, proprietary

Orig-Change-Id: Ic0a0f2a78b9d0705b58b98f340a29cb597ae15fe

Change-Id: I3fc5ebe8134c21b1c23eeb0cbd95080db1669bb7
Signed-off-by: Charitardha <satya.charitardha.jayanti@intel.com>
Reviewed-on: http://android.intel.com:8080/80765
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 1d8feac..f14cc44 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -2368,12 +2368,20 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
          return OK;
     }
 #else
-    if (objectTypeIndication  == 0x6b) {
+    if (objectTypeIndication  == 0x6b || objectTypeIndication == 0x69) {
+#ifdef USE_INTEL_MDP
+        // This is mp3 audio
+        // Google decoder was not able to handle packetized MP3 audio.
+        // Intel CIP decoder supports and hence the change.
+        // This need not be upstreamed to AOSP, but changes needed in Intel software base.
+        mLastTrack->meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_AUDIO_MPEG);
+#else
         // The media subtype is MP3 audio
         // Our software MP3 audio decoder may not be able to handle
         // packetized MP3 audio; for now, lets just return ERROR_UNSUPPORTED
         ALOGE("MP3 track in MP4/3GPP file is not supported");
         return ERROR_UNSUPPORTED;
+#endif
     }
 #endif
 
-- 
1.7.4.1

